src/App/Repository/ConfigRepository.php

```diff
@@ -104,6 +119,23 @@ public function update(Entity $entity)
 		}
 	}
 
+	private function insertOrUpdate($group, $key, $value)
+	{
+		$value = json_encode($value);
+
+		try {
+			DB::insert('config', array('group_name', 'config_key', 'config_value'))
+				->values(array($group, $key, $value))
+				->execute($this->db);
+		} catch (\Database_Exception $e) {
```

Comment:
From @willdoran comment:

> I think doing away with the generic json storage for dataproviders would be preferable. It would kind of make sense if we had dozens of dataproviders but in reality we don't and they are never going to increase in number at a particularly dramatic rate. Specifically, for data providers I would think we would be best to add a dataprovider archetype and then extend for that into SMS types, email, twitter etc.

I'm somewhat divided about this. While explicit is generally good 👍 , I think we should also bear in mind the custom integration needs that may appear in individual projects we do for clients. 

Creating specific archetypes for those may make it more difficult to merge in updates from main platform into those project repositories?